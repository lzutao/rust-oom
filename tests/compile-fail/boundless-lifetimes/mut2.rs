fn main() {
    let mut t = 0;
    let slice = oom::NonEmptySlice::<u8>::from_mut(&mut t);
    let first = slice.first_mut();
    let second = slice.first_mut();
    assert_eq!(first as *mut _, second as *mut _);
    *first = 0;
    *second = 1;
    assert_eq!(*first, 1, "UB, written here through second.");
}
