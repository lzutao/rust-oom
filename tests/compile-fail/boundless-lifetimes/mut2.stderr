error[E0499]: cannot borrow `*slice` as mutable more than once at a time
 --> $DIR/mut2.rs:5:18
  |
4 |     let first = slice.first_mut();
  |                 ----- first mutable borrow occurs here
5 |     let second = slice.first_mut();
  |                  ^^^^^ second mutable borrow occurs here
6 |     assert_eq!(first as *mut _, second as *mut _);
  |                ----- first borrow later used here
